### DISCUSSION TITLE
Using an Azure DevOps pipeline to deploy web app

### DISCUSSION THREAD

**Discussion by ninyes** [microsoft/sample-app-aoai-chatGPT#466] [Q&A]
Hi all,

I recently deployed an Azure OpenAI web app and I want to customize the frontend. Therefore, I have connected the web app to a branch in the DevOps repo (cloned the sample app repo) and I manually run the AZD commands to build the frontend in VSCode. After this I deploy the web app in the deployment center. 

Preferably, I would like to create an Azure DevOps pipeline that starts the deployment (and if possible build) process once a commit is made to a branch (e.g. "main"). I have tried various YAML configurations, but for some reason nothing seems to work as intended. Who has managed to get this to work and willing to share their _azure-pipelines.yml_ file?

Many thanks in advance!

**Comment by ninyes** [comment]
In case anyone has the same issue: 
I managed to do this by using the [python-to-linux-webapp](https://github.com/microsoft/azure-pipelines-yaml/blob/master/templates/python-to-linux-webapp-on-azure.yml) template. Before the archiving step in the build stage, I have added the following to actually build the frontend:

- task: NodeTool@0
  inputs:
    versionSpec: '20.x'
  displayName: 'Install Node.js'

- script: |
    cd ./frontend
    npm install
    npm run build
    cd ..
    rm -rf $(projectRoot)/frontend/node_modules
  displayName: 'npm install and build, removing node_modules'
 
Hope this helps!

**Reply by ghost** [reply]
Can you please share your pipeline yml file :)
